src/utils_tail_match.c

```diff
@@ -118,10 +118,10 @@ static int latency_submit_match(cu_match_t *match, void *user_data) {
   sstrncpy(vl.type, data->type, sizeof(vl.type));
   for (size_t i = 0; i < data->latency_config.percentile_num; i++) {
     if (strlen(data->type_instance) != 0)
-      snprintf(vl.type_instance, sizeof(vl.type_instance), "%.117s-%.2f",
+      snprintf(vl.type_instance, sizeof(vl.type_instance), "%.50s-%.4g",
```

Comment:
This format was originally `"%s-%.0f"`. Can we go back to using `%.0f` again?

Reply:
As stated before, that is done to allow to report percentiles with fractional values.
```
# perl -e 'use POSIX; printf("%.0f\n", 99.5);'
100
# perl -e 'use POSIX; printf("%.4g\n", 99.5);'
99.5
# perl -e 'use POSIX; printf("%.4g\n", 99);'
99
```
Configuration allows to set fractional percentile values, so we need to allow report them correctly.
If you want, we can split this change - set `%.0f` in 5.8 and `%.4g` in master.

Reply:
The docs don't say that fractional percentiles are forbidden, so it makes sense to support them. Put the `%g` variant into 5.8.

Maybe we can increase the precision to `%.5g` though – somebody is going to want to track five nines … ;)